### DISCUSSION TITLE
Installation of Omnetpp-5.7   on centos 7.9 failed

### DISCUSSION THREAD

**Discussion by tmaria2** [omnetpp/omnetpp#1031] [Q&A] [answered]
Hello

I am trying to install Omnetpp-5.7 on centos 7,9.
During the .configure, it fails with error "configure: error: Please adjust CXXFLAGS in configure.user to enable C++14 compilation or install a C++14-capable compiler (e.g. clang 3.4+ or gcc 6.1+)"
Centos 7, has gcc 4.8.5 installed, which does not seems to support C++14
Has anyone installed  successfully Omnetpp-5.7 on centos 7,9?

Best  regards

Maria

**Comment by rhornig** [comment]
You can install newer versions of development tool that work. `devtoolset-10` is known to work, but the IDE will not function either as the glibc libarary on CentOS7 is just too old. CentOS7 is just too old and EoL.

**Comment by tmaria2** [comment]
Hello Rudolf,

Thank you for your prompt reply
We will upgrade our labs to Rocky in the near future, but at the moment we remain with Centos 7. Which version of omnetpp is fully supported (including IDE) by Centos 7?

Best regards

Maria

------------------------------------------------------------------------------------------------------------------------
Maria Tsiolakki-Papandreou
Senior Assistant IT Officer
Computer Science Department   |  University of Cyprus
(dir no:  ++357 22892727  /  fax no:  ++357 22892701 /   email:  Maria.Tsiolakki_AT_cs.ucy.ac.cy)
------------------------------------------------------------------------------------------------------------------------

**Reply by rhornig** [reply]
I would try, OMNeT++ 5.6 or 5.7. 

I have reports that `devtoolset-10` can be used to build omnet 6 on CetnOS 7. If needed, I caan also provide a binary patched shared library that make the IDE working on CentOS 7. Let me know if you need the IDE patch.

**Comment by tmaria2** [comment]
Hello Rudolf,

Yes of course! Please sent me with all the details for the installation. So the patch will be for omnet 6 on CentosOS 7 not  for OMNeT++ 5.6 or 5.7. on centos 7, right?

Best regards

Maria

------------------------------------------------------------------------------------------------------------------------
Maria Tsiolakki-Papandreou
Senior Assistant IT Officer
Computer Science Department   |  University of Cyprus
(dir no:  ++357 22892727  /  fax no:  ++357 22892701 /   email:  Maria.Tsiolakki_AT_cs.ucy.ac.cy)
------------------------------------------------------------------------------------------------------------------------

**Reply by rhornig** [reply, marked as answer]
Yes. For OMNeT++ 6.0.1, install devtoolset-1

    $ yum install devtoolset-10

activate the toolset in a new bash shell (you MUST work from that shell):

    $ scl enable devtoolset-10 'bash'

That will open a new bash shell with all the compilers and tools activated. Always use that shell to `source setenv` and the configure and build OMNeT++.

To patch the IDE: 

You should replace the liboppsim.so file in `ide/plugins/org.omnetpp.ide.nativelibs.linux.x86_64_6.0.1.220905-55d93ce6e9`
with the new one.

Download from here:
https://drive.google.com/drive/u/1/folders/1zafJlQl-skqsTIzM62TSPzYLUA7runNe
